### PR TITLE
update maven android api level

### PR DESCRIPTION
**Pull request by timwr**
This pr updates the android api level so we can use more recent Android apis.
Also includes a minor fix to the java version in the android pom.xml so that Intellij no longer complains about the @Overrides.

### CODE REVIEW

**Comment by bcook-r7** [comment]
so, this is the death of Android 2.3 support then? we talked about this last year, it being on its way out then. Is there a sufficiently-small user-base now that this is worth doing?

File: java/androidpayload/library/pom.xml [hunk 1]
```diff
@@ -59,12 +59,21 @@
 				<artifactId>android-maven-plugin</artifactId>
 				<configuration>
 					<sdk>
-						<!-- platform or api level (api level 10 = platform 2.3)-->
-						<platform>10</platform>
+						<!-- platform or api level (api level 19 = platform 4.4 (kitkat))-->
```
Comment (bcook-r7) [inline java/androidpayload/library/pom.xml]: 4.4 or 4.1.1.4 ?

**Comment by bcook-r7** [comment]
we'll probably need to add the appropriate line to here to pull the right version into the build system:

https://github.com/rapid7/metasploit-payloads/blob/master/docker/Dockerfile#L38

Are you docker-savvy / would you mind adding that to the PR too?

File: java/androidpayload/library/pom.xml (hunk 1 above)
Comment (timwr) [inline java/androidpayload/library/pom.xml]: 4.1.1.4 is the maven plugin version, 4.4 is the android sdk version which is OK here.

**Comment by timwr** [comment]
We're actually still supporting 2.3, as long as we don't use the new API's without checking.
I'll double check the clipboard apis are being loaded lazily so 2.3 should still work fine. This change just means the newer API's are available. We still support 2.3 unless we change this: https://github.com/rapid7/metasploit-payloads/blob/master/java/androidpayload/app/src/main/AndroidManifest.xml#L7

I'll update the Dockerfile :)

**Comment by bcook-r7** [comment]
Subsequent PRs should pass CI now that the docker image is updated.

**Pull request merged**